ape_ens/converter.py

```diff
@@ -1,15 +1,34 @@
-from contextlib import contextmanager
-from typing import Any
+from typing import Any, Dict
 
 from ape.api import ConverterAPI, Web3Provider
 from ape.exceptions import NetworkError, ProviderError
 from ape.types import AddressType
+from ape.utils import cached_property
 from web3.main import ENS
 
 
 class ENSConversions(ConverterAPI):
     """Converts ENS names like `my-name.eth` to `0xAbCd...1234`"""
 
+    address_cache: Dict[str, AddressType] = {}
+
+    @cached_property
+    def mainnet_provider(self) -> Web3Provider:
+        provider = self.network_manager.active_provider
+        if (
+            not provider
+            or not isinstance(provider, Web3Provider)
+            or not provider.network.name == "mainnet"
+        ):
+            provider = self.network_manager.get_provider_from_choice("ethereum:mainnet")
+
+            if not isinstance(provider, Web3Provider):
+                raise ValueError("Must use a Web3Provider for default mainnet provider.")
+
+            provider.connect()
```

Comment:
Might want to try..except here on if it fails to connect, with an error message that says "hey, make your default provider for eth mainnet into something that will work" because the default is geth